js/woo.js

```diff
@@ -0,0 +1,1465 @@
+        }
+        const addressTo = this.safeString (transaction, 'target_address');
+        const addressFrom = this.safeString (transaction, 'source_address');
+        const timestamp = this.safeTimestamp (transaction, 'created_time');
+        return {
+            'id': this.safeString (transaction, 'id'),
+            'txid': this.safeString (transaction, 'tx_id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'address': movementDirection === 'deposit' ? addressTo : addressFrom, // TODO - should become deprecated property
+            'address_from': addressTo, // TODO - make unified property
+            'address_to': addressFrom, // TODO - make unified property
+            'tag': this.safeString (transaction, 'extra'),
+            'type': movementDirection,
+            'amount': this.safeNumber (transaction, 'amount'),
+            'currency': code,
+            'status': this.parseTransactionStatus (this.safeString (transaction, 'status')),
+            'updated': this.safeTimestamp (transaction, 'updated_time'),
+            'fee': {
+                'currency': fee_token,
+                'cost': this.safeNumber (transaction, 'fee_amount'),
+                'rate': undefined,
+            },
+            'info': transaction,
+        };
+    }
+
+    parseTransactionStatus (status) {
+        const statuses = {
+            'NEW': 'pending',
+            'CONFIRMING': 'pending',
+            'PROCESSING': 'pending',
+            'COMPLETED': 'ok',
+            'CANCELED': 'canceled',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    async fetchTransfers (code = undefined, since = undefined, limit = undefined, params = {}) {
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+        }
+        const request = {
+            'type': 'COLLATERAL',
+        };
+        params = this.omit (params, 'type'); // conflict! we need to change 'type' param for spot/futures into 'exchangeType'
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchTransfers', undefined, params);
+        const method = this.getSupportedMapping (marketType, {
+            'spot': 'v1PrivateGetAssetHistory',
+        });
+        const response = await this[method] (this.extend (request, query));
+        const rows = this.safeValue (response, 'rows');
+        return this.parseTransfers (rows, currency, since, limit, params);
+    }
+
+    parseTransfer (transfer, currency = undefined) {
+        // example is "fetchTransactions"
+        let movementDirection = this.safeStringLower (transfer, 'token_side');
+        if (movementDirection === 'withdraw') {
+            movementDirection = 'withdrawal';
+        }
+        let fromAccount = undefined;
+        let toAccount = undefined;
+        if (movementDirection === 'withdraw') {
+            fromAccount = 'derivative';
```

Comment:
We should not set it to the word `derivative`, since it's nowhere to find in any other places through this exchange and it's not unified.

Reply:
Should I set it to 'derivatives' (as in bitfinex) or 'futures' ?  because Woo api docs talks about collaterals and they are probably going to open the futures/derivs . however, what I should do in this case so?

Reply:
We are in the process of unifying it in the following way: https://github.com/ccxt/ccxt/pulls?q=fetchmarkets+unified+is%3Apr

- to `exchange.has` we add boolean flags: `spot`, `swap`, `future`, `option`
- to the market structure we add boolean flags: `spot`, `swap`, `future`, `option` and we also add `type` string (equal to either of `'spot'`, `'swap'`, `'future'`, `'option'`) and a `contract` boolean flag that is set to `swap || future || option`.